go.mod

```diff
@@ -5,6 +5,7 @@ go 1.13
 require (
 	github.com/containous/traefik/v2 v2.1.2
 	github.com/coreos/go-oidc v2.1.0+incompatible
+	github.com/nirasan/go-oauth-pkce-code-verifier v0.0.0-20170819232839-0fbfe93532da
```

Comment:
Is there an alternative which is more popular?

Reply:
Maybe, I found it in https://github.com/asaintsever/traefik-forwardauth-authcode-pkce/blob/master/src/main.go#L17

Reply:
Just google  "go pkce", "go oauth pkce" and found the repo on top 3

Reply:
Using some random repo for security related things is not a smart idea. If there is a good alternative we should use that one.

Reply:
I agree, I'l try to find something else

Reply:
@SuperSandro2000 , what about this one https://github.com/lyokato/goidc/tree/master/pkce ?

Reply:
https://github.com/nirasan/go-oauth-pkce-code-verifier - there is just one file [verifier.go](https://github.com/nirasan/go-oauth-pkce-code-verifier/blob/master/verifier.go) with minimal code.  
I think we can trust the repo, at least I've made CR and everything is fine